subscribers.js

```diff
@@ -1,7 +1,95 @@
 "use strict";
 
-// Set of all registered emails.
-// TODO: Implement a real persistent storage solution.
-const gEmails = new Set();
+const pg = require("pg");
 
-module.exports = gEmails;
+const AppConstants = require("./app-constants");
+
+const dbConfig = {
+  database: AppConstants.DATABASE_NAME,
+  host: AppConstants.DATABASE_HOST,
+  port: AppConstants.DATABASE_PORT,
+  max: 10,
+};
+
+const dbPool = new pg.Pool(dbConfig);
+
+function dbq(aQuery) {
+  return dbPool.query(aQuery);
+}
+
+function _ret(aSuccess, aError, aOther) {
+  return Object.assign({
+    success: aSuccess,
+    error: aError || null,
+  }, aOther);
+}
+
+const Subscribers = {
+  async addUser(aEmail) {
+    // Error code for attempting to add a duplicate entry on a UNIQUE key.
+    const DUPLICATE_ERROR = 23505;
+    try {
+      await dbq({
+        text: "INSERT INTO users ( email ) VALUES ( $1 );",
+        values: [ aEmail ],
+      });
+
+      return _ret(true);
+    } catch (e) {
+      if (e.code == DUPLICATE_ERROR) {
+        // Duplicate entry, count it as success.
+        return _ret(true, null, { duplicate: true });
+      }
+
+      return _ret(false, e);
+    }
+  },
+
+  async deleteUser(aEmail) {
+    try {
+      await dbq({
+        text: "DELETE FROM users WHERE email = $1;",
+        values: [ aEmail ],
+      });
+      return _ret(true);
+    } catch(e) {
+      return _ret(false, e);
+    }
+  },
+
+  async getUser(aEmail) {
+    try {
+      const rows = (await dbq({
+        text: "SELECT * FROM users WHERE email = $1;",
+        values: [ aEmail ],
+      })).rows;
+
+      if (!rows.length) {
+        return _ret(false);
+      }
+
+      const { email } = rows[0];
+      if (!email) {
+        // This shouldn't happen, but... better safe than sorry.
+        return _ret(false, "Entry was found but with no value");
```

Comment:
In the `try..catch` blocks (see below L79, etc), we do `return _ret(false, e)` to return an error. Should this `aError` string be wrapped in an Error object too?
For example:
```js
return _ret(false, new Error("Entry was found but with no value"));
```

Not sure if I've grokked enough of the API code yet to see how we're handling the errors in Express yet to see how it'd handle strings vs objects or if these error messages aren't user facing yet.